apps/meteor/server/models/raw/Users.js

```diff
@@ -258,6 +258,112 @@ export class UsersRaw extends BaseRaw {
 		return this.findPaginated(query, options);
 	}
 
+	findPaginatedActiveUsersByRoomIdWithHighestRole(
+		searchTerm,
+		rid,
+		options,
+		searchFields,
+		extraQuery = [],
+		{ startsWith = false, endsWith = false } = {},
+	) {
+		if (options == null) {
+			options = {};
+		}
+
+		const termRegex = new RegExp((startsWith ? '^' : '') + escapeRegExp(searchTerm) + (endsWith ? '$' : ''), 'i');
+
+		const orStmt = (searchFields || []).reduce(function (acc, el) {
+			acc.push({ [el.trim()]: termRegex });
+			return acc;
+		}, []);
+
+		const query = {
+			$and: [
+				{
+					active: true,
+					__rooms: rid,
+					username: { $exists: true },
+					// if the search term is empty, don't need to have the $or statement (because it would be an empty regex)
+					...(searchTerm && orStmt.length > 0 && { $or: orStmt }),
+				},
+				...extraQuery,
+			],
+		};
+
+		const skip =
+			options.skip !== 0
+				? [
+						{
+							$skip: options.skip,
+						},
+				  ]
+				: [];
+
+		const limit =
+			options.limit !== 0
+				? [
+						{
+							$limit: options.limit,
+						},
+				  ]
+				: [];
+
+		return this.col.aggregate([
+			{
+				$match: query,
+			},
+			{
+				$lookup: {
+					from: 'rocketchat_subscription',
+					localField: '_id',
+					foreignField: 'u._id',
+					as: 'sub',
+					pipeline: [
+						{
+							$match: {
+								$expr: {
+									$eq: ['$rid', rid],
+								},
+							},
+						},
+					],
+				},
+			},
+			{ $unwind: '$sub' },
+			{
+				$project: {
+					...options.projection,
+					roles: '$sub.roles',
+				},
+			},
+			{
+				$addFields: {
+					highestRole: {
+						$cond: [{ $in: ['owner', '$roles'] }, 'owner', { $cond: [{ $in: ['moderator', '$roles'] }, 'moderator', 'member'] }],
```

Comment:
A suggestion: instead of doing the same condition twice to check highest and level, why not making this one return an object like { role: owner, level: 1 }

Reply:
Ohh I agree it'd look better. Update, tkss :)